### ISSUE TITLE
Gap where titlebar used to be when starting maximised app

### ISSUE THREAD

**Issue by lightbulbjim** [velitasali/GTKTitleBar#15]
When a maximised app is closed and reopened there is sometimes a gap where the titlebar would otherwise be. Unmaximising and remaximising the app fixes it so that the gap is gone.

Steps to reproduce:

1. Launch an app with a titlebar.
2. Maximise the app. The titlebar is now hidden.
3. Close the app while maximised.
4. Relaunch the app.

Since it's intermittent, maybe it's a race condition between removing the window decoration and resizing the window?

Screenshot showing the problem:
![Screenshot from 2019-06-19 10-26-47](https://user-images.githubusercontent.com/4312404/59728400-38061200-927d-11e9-97ee-61eef7cd9b51.png)

**Comment by lightbulbjim** [comment]
Note that the gap also shows up in the overview:
![Screenshot from 2019-06-19 10-37-34](https://user-images.githubusercontent.com/4312404/59728663-5ddfe680-927e-11e9-836e-5a624a0130cd.png)

**Comment by velitasali** [comment]
@lightbulbjim Could this be caused by some other extension? You are the only one reporting this issue.

**Comment by lightbulbjim** [comment]
I just tested with all extensions other than GTKTitleBar disabled and it happened again.

I forgot to mention what versions I'm running; I'm using GTKTitleBar 3.0 and Gnome 3.32.2.

**Comment by velitasali** [comment]
I am not sure what is causing this. Are you using a custom theme?
![Screenshot from 2019-07-06 13-39-12](https://user-images.githubusercontent.com/10258162/60755102-7fb5c900-9ff3-11e9-877e-306a0169a5a0.png)

**Comment by lightbulbjim** [comment]
No, I'm using vanilla Adwaita.

**Comment by velitasali** [comment]
> No, I'm using vanilla Adwaita.

Hey, after reading the description again, I realized I might have the same issue but it goes away after the window is focused. Is that what you are experiencing?

**Comment by lightbulbjim** [comment]
No, the gap remains when the window is focused. Doing an action which redraws the window (eg unmaximising/maximising) makes the gap disappear.

I haven't experienced the problem for the last week or so. It's possible that some other package update has fixed it.

**Comment by velitasali** [comment]
> No, the gap remains when the window is focused. Doing an action which redraws the window (eg unmaximising/maximising) makes the gap disappear.
> 
> I haven't experienced the problem for the last week or so. It's possible that some other package update has fixed it.

Hm, I think I still have it. Which version of GNOME are you running?

**Comment by lightbulbjim** [comment]
3.32.2

**Comment by velitasali** [comment]
On Ubuntu 19.04, I have 3.32.1, so like you said it might be something that was fixed in 3.32.2.

**Comment by parcelcat** [comment]
I see the same thing in Visual Studio Code in GNOME 3.32.2 about half the time the application is launched. I have the menu bar hidden in Visual Studio Code. From the screenshots, it looks like @lightbulbjim also has the menu bar hidden while @velitasali does not.

**Comment by lightbulbjim** [comment]
I don’t think it’s related to the menu bar. I also saw the problem on apps which don’t have a menu bar at all, for example OpenRCT2. 

My impression was that the viewport was being resized before the titlebar was hidden. So the result looked like the titlebar disappeared but the space it left wasn’t filled.
